src/Microsoft.AspNetCore.Razor/Parser/TagHelpers/TagHelperBlockRewriter.cs

```diff
@@ -347,6 +370,28 @@ public static class TagHelperBlockRewriter
             // Have a name now. Able to determine correct isBoundNonStringAttribute value.
             var result = CreateTryParseResult(name, descriptors);
 
+            var firstChild = builder.Children[0] as Span;
+            if (firstChild != null && firstChild.Symbols[0] is HtmlSymbol)
+            {
+                var htmlSymbol = firstChild.Symbols[firstChild.Symbols.Count - 1] as HtmlSymbol;
+                switch (htmlSymbol.Type)
+                {
+                    // We purposefully do not capture NoQuotes at parse time to protect users from rendering
+                    // dynamic content with spaces that can break attributes.
+                    // Ex: <a my-attribute=@value /> where @value results in the test "hello world".
```

Comment:
This is the same example as in the above comment. The example likely matches this case. If that's correct, the earlier comment should say something about _not_ special-casing `my-attribute=literal-value` for consistency and for simplifying later code.